packages/hardhat-core/src/internal/solidity/compiler/index.ts

```diff
@@ -17,32 +18,21 @@ export class Compiler implements ICompiler {
   public async compile(input: CompilerInput) {
     const scriptPath = path.join(__dirname, "./solcjs-runner.js");
 
-    const output: string = await new Promise((resolve, reject) => {
-      try {
-        const subprocess = execFile(
-          process.execPath,
-          [scriptPath, this._pathToSolcJs],
-          {
-            maxBuffer: 1024 * 1024 * 500,
-          },
-          (err, stdout) => {
-            if (err !== null) {
-              return reject(err);
-            }
-            resolve(stdout);
-          }
-        );
-
-        subprocess.stdin!.write(JSON.stringify(input));
-        subprocess.stdin!.end();
-      } catch (e: any) {
-        throw new HardhatError(
-          ERRORS.SOLC.SOLCJS_ERROR,
-          { error: e.message },
-          e
-        );
-      }
-    });
+    let output: string;
+    try {
+      const { stdout } = await execFileWithInput(
+        process.execPath,
+        [scriptPath, this._pathToSolcJs],
+        JSON.stringify(input),
+        {
+          maxBuffer: 1024 * 1024 * 500,
+        }
+      );
+
+      output = stdout;
+    } catch (e: any) {
+      throw new HardhatError(ERRORS.SOLC.CANT_RUN_NATIVE_COMPILER, {}, e);
```

Comment:
Should this be `ERRORS.SOLC.SOLCJS_ERROR`?

Reply:
Good catch!

Reply:
@alcuadrado I think the error types are swapped between socljs and native implementations :sweat_smile: